AnnService/src/Core/SPANN/SPANNIndex.cpp

```diff
@@ -212,7 +212,7 @@ namespace SPTAG
                 workSpace->m_postingIDs.clear();
 
                 float limitDist = p_queryResults->GetResult(0)->Dist * m_options.m_maxDistRatio;
-                for (int i = 0; i < m_options.m_searchInternalResultNum; ++i)
+                for (int i = 0; i < min(m_options.m_searchInternalResultNum, p_queryResults->GetResultNum()); ++i)
```

Comment:
We may need to enlarge the queryset before the headindex search? Otherwise we cannot increase the internalresultnum to get better recall.

Reply:
We will segfault if p_queryResults is smaller than searchInternalResultNum

Maybe it is better to just iterate over p_queryResults->GetResultNum()?

Reply:
no, it should be searchInternalResultNum. We need to ensure the GetResultNum() is equal or larger than searchInternResultNum.

Reply:
I agree that callers should provide `QueryResultSet`s with more than `searchInternalResultNum` results, which we enforce in our `SSDServing`. 

However, if we get to this point with the result set being too small, I think it is better to complete the search than to segfault. I can add a debug assert if you would like to warn the caller about it